pkg/cmd/cli/cmd/loginoptions.go

```diff
@@ -222,8 +222,9 @@ func (o *LoginOptions) gatherAuthInfo() error {
 		}
 	}
 
-	// if a username was provided try to make use of it
-	if o.usernameProvided() {
+	// if a username was provided try to make use of it, but if a password were provided we force a token
+	// request which will return a proper response code for that given password
+	if o.usernameProvided() && !o.passwordProvided() {
```

Comment:
When password were explicitly provided through `--password` we are supposed to try to authenticate instead of reusing a matching stanza, because for example the password provided can be invalid.